src/boot/efi/boot.c

```diff
@@ -2707,9 +2728,9 @@ static EFI_STATUS run(EFI_HANDLE image) {
                                 "No loader found. Configuration files in \\loader\\entries\\*.conf are needed.");
 
         /* select entry or show menu when key is pressed or timeout is set */
-        if (config.force_menu || config.timeout_sec > 0)
+        if (config.force_menu || (config.timeout_sec > 0 && config.timeout_sec != TIMEOUT_MENU_DISABLED))
```

Comment:
might as well clean this up then to turn it into:

`if (config.force_menu || !IN_SET(config.timeout_sec, TIMEOUT_MENU_HIDDEN, TIMEOUT_MENU_DISABLE))`